### PR TITLE
Add form-based optional styling

### PR DESCRIPTION
**Pull request by TheTallJerry**
- Adds a collapsable accordion/pane of two config options: a numerical seed and a boolean automation. 
- Tests and some non-breaking refactoring.
Ex. 1 
image
Ex. 2
image
Ex. 3
image

### CODE REVIEW

File: demo/src/MemoryModelsUserInput.tsx [hunk 1]
```diff
@@ -114,6 +133,63 @@ function MemoryModelsTextInput(props: MemoryModelsTextInputPropTypes) {
...
+                    sx={{ width: "50%" }}
+                />
+                <FormControlLabel
+                    control={
+                        <Checkbox
+                            checked={props.configData.useAutomation}
+                            onChange={handleAutomationChange}
+                        />
+                    }
+                    label="Generate with automation"
```
Comment (david-yz-liu) [inline demo/src/MemoryModelsUserInput.tsx:186]: More accurate label is "Use automatic layout"

File: demo/src/MemoryModelsUserInput.tsx [hunk 2]
```diff
@@ -122,19 +198,33 @@ export default function MemoryModelsUserInput(
...
+                        <MemoryModelsTextInput
+                            textData={props.textData}
+                            setTextData={props.setTextData}
+                        />
+                        <Accordion>
+                            <AccordionSummary
+                                expandIcon={<ExpandMore />}
+                                data-testid="optional-styles-accordion"
+                            >
+                                Optional Styling
```
Comment (david-yz-liu) [inline demo/src/MemoryModelsUserInput.tsx:217]: A more accurate name here is "Rendering Options"

File: demo/src/MemoryModelsUserInput.tsx [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
 } from "@mui/material";
 import DownloadJSONButton from "./DownloadJSONButton";
+import { ExpandMore } from "@mui/icons-material";
+
+interface configDataPropTypes {
+    useAutomation: boolean;
+    overallDrawConfig: {
+        seed: Number;
+    };
+    individualDrawConfig: Array<any>;
```
Comment (david-yz-liu) [inline demo/src/MemoryModelsUserInput.tsx:25]: this isn't being used, and should be removed from the state

File: demo/src/__tests__/MemoryModelsUserInput.spec.tsx [hunk 4]
```diff
@@ -174,4 +213,71 @@ describe("MemoryModelsUserInput", () => {
...
+                    configData={configDataMock}
+                    setConfigData={setConfigDataMock}
+                />
+            );
+        });
+
+        it("renders a number input with correct props and checkbox that is checked by default", () => {
+            const seedInput: HTMLInputElement =
+                screen.getByTestId("config-seed");
+            //TODO: get max and min values from backend config
```
Comment (david-yz-liu) [inline demo/src/__tests__/MemoryModelsUserInput.spec.tsx:235]: Remove this comment

File: demo/src/__tests__/MemoryModelsUserInput.spec.tsx [hunk 5]
```diff
@@ -174,4 +213,71 @@ describe("MemoryModelsUserInput", () => {
...
+        });
+
+        it("handles seed change", () => {
+            const seedInput: HTMLInputElement =
+                screen.getByTestId("config-seed");
+            const mockSeed = "123";
+            fireEvent.change(seedInput, { target: { value: mockSeed } });
+            expect(setConfigDataMock).toHaveBeenNthCalledWith(
+                1,
+                expect.objectContaining({
```
Comment (david-yz-liu) [inline demo/src/__tests__/MemoryModelsUserInput.spec.tsx:259]: I'm not sure why you're using `objectContaining` instead of `toEqual` here (and elsewhere)?

Comment (TheTallJerry) [reply demo/src/__tests__/MemoryModelsUserInput.spec.tsx:259]: Good catch, used `objectContaining` initially when the order wasn't preserved, fixed

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**